doc/source/batch-inference/preprocessing.rst

```diff
@@ -0,0 +1,118 @@
+~~~~~~~~~~~~~~
+The batches that your UDF accepts can be different formats depending on the `batch_format` that is passed to :meth:`ds.map_batches() <ray.data.Dataset.map_batches>`
+
+Here is an overview of the available batch formats:
+
+.. tabbed:: "default"
+
+  The "default" batch format presents data as follows for each Dataset type:
+
+  * **Tabular Datasets**: Each batch will be a
+    `pandas.DataFrame <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`__.
+    This may incur a conversion cost if the underlying Dataset block is not
+    zero-copy convertible from an Arrow table.
+
+    .. literalinclude:: ../data/doc_code/transforming_datasets.py
+      :language: python
+      :start-after: __writing_default_udfs_tabular_begin__
+      :end-before: __writing_default_udfs_tabular_end__
+
+  * **Tensor Datasets** (single-column): Each batch will be a single
+    `numpy.ndarray <https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html>`__
+    containing the single tensor column for this batch.
+
+    .. literalinclude:: ../data/doc_code/transforming_datasets.py
+      :language: python
+      :start-after: __writing_default_udfs_tensor_begin__
+      :end-before: __writing_default_udfs_tensor_end__
+
+  * **Simple Datasets**: Each batch will be a Python list.
+
+    .. literalinclude:: ../data/doc_code/transforming_datasets.py
+      :language: python
+      :start-after: __writing_default_udfs_list_begin__
+      :end-before: __writing_default_udfs_list_end__
+
+.. tabbed:: "pandas"
+
+  The ``"pandas"`` batch format presents batches in
+  `pandas.DataFrame <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`__
+  format. If converting a simple dataset to Pandas DataFrame batches, a single-column
+  dataframe with the column ``"__value__"`` will be created.
+
+  .. literalinclude:: ../data/doc_code/transforming_datasets.py
+    :language: python
+    :start-after: __writing_pandas_udfs_begin__
+    :end-before: __writing_pandas_udfs_end__
+
+.. tabbed:: "numpy"
+
+  The ``"numpy"`` batch format presents batches in
+  `numpy.ndarray <https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html>`__
+  format as follows:
+
+  * **Tabular Datasets**: Each batch will be a dictionary of NumPy
+    ndarrays (``Dict[str, np.ndarray]``), with each key-value pair representing a column
+    in the table.
+
+  * **Tensor Datasets** (single-column): Each batch will be a single
+    `numpy.ndarray <https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html>`__
+    containing the single tensor column for this batch.
+
+  * **Simple Datasets**: Each batch will be a single NumPy ndarray, where Datasets will
+    attempt to convert each list-batch to an ndarray.
+
+  .. literalinclude:: ../data/doc_code/transforming_datasets.py
+    :language: python
+    :start-after: __writing_numpy_udfs_begin__
+    :end-before: __writing_numpy_udfs_end__
+
+
+Configuring Batch Size
+~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Ah I see you have it here. IMO, this page should be combined with the former, since preprocessing + inference are so tied together. Also, batch size applies for both preprocessing and inference.